test/src/telemetry/test_telemetry.c

```diff
@@ -82,9 +82,11 @@ test_factory(ConnectionType type, int status, char *host, int port)
 		ts_connection_mock_set_recv_buf(conn, test_string, strlen(test_string));
 #endif
 
-	int retries = 0;
-	while (retries < INVALID_STATUS_RETRIES)
+	int retries = -1;
+	while (retries < INVALID_RESPONSE_RETRIES)
 	{
+		retries++;
```

Comment:
Just an alternative. Ignore if you like.

```suggestion
	for (int retries = -1; retries < INVALID_RESPONSE_RETRIES; retries++)
	{
```